aws/resource_aws_client_vpn_endpoint_test.go

```diff
@@ -0,0 +1,259 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAwsClientVpnEndpoint_importBasic(t *testing.T) {
+	resourceName := "aws_client_vpn_endpoint.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProvidersWithTLS,
+		CheckDestroy: testAccCheckAwsClientVpnEndpointDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccClientVpnEndpointConfig(acctest.RandString(5)),
+			},
+
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAwsClientVpnEndpoint_basic(t *testing.T) {
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProvidersWithTLS,
+		CheckDestroy: testAccCheckAwsClientVpnEndpointDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccClientVpnEndpointConfig(acctest.RandString(5)),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsClientVpnEndpointExists("aws_client_vpn_endpoint.test"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAwsClientVpnEndpoint_withLogGroup(t *testing.T) {
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProvidersWithTLS,
+		CheckDestroy: testAccCheckAwsClientVpnEndpointDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccClientVpnEndpointConfigWithLogGroup(acctest.RandString(5)),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsClientVpnEndpointExists("aws_client_vpn_endpoint.test"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAwsClientVpnEndpoint_withDNSServers(t *testing.T) {
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProvidersWithTLS,
+		CheckDestroy: testAccCheckAwsClientVpnEndpointDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccClientVpnEndpointConfigWithDNSServers(acctest.RandString(5)),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsClientVpnEndpointExists("aws_client_vpn_endpoint.test"),
+				),
+			},
+		},
+	})
+}
+
```

Comment:
There aren't any test cases here that test updating the resource in `resourceAwsClientVpnEndpointUpdate`

Reply:
I've rolled this into the `withLogGroup` and `withDNSServers` tests.  Two birds with one stone.